src/VisualStudio/LiveShare/Impl/RunCodeActionsHandler.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.LanguageServer.CustomProtocol;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Microsoft.VisualStudio.LiveShare.LanguageServices;
+using Newtonsoft.Json.Linq;
+using LSP = Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.VisualStudio.LanguageServices.LiveShare
+{
+    /// <summary>
+    /// Run code actions handler.  Called when lightbulb invoked.
+    /// Code actions must be applied from the UI thread in VS.
```

Comment:
can we assert this is happening?

Reply:
This is asserted when it tries to apply the code action operation (how I noticed it was broken).  Not sure if it is worth asserting here as well?

Reply:
There's nothing really to _assert_ -- this is doing JTF style switching so it just switches when it needs it.